src/lib/operators/projection.cpp

```diff
@@ -97,47 +97,109 @@ std::shared_ptr<const Table> Projection::_on_execute() {
     step_performance_data.set_step_runtime(OperatorSteps::UncorrelatedSubqueries, timer.lap());
   }
 
+  auto forwarding_cost = std::chrono::nanoseconds{};
+  auto expression_evaluator_cost = std::chrono::nanoseconds{};
+
+  const auto chunk_count = input_table.chunk_count();
+
   // Perform the actual projection on a per-chunk level. `output_segments_by_chunk` will contain both forwarded and
   // newly generated columns. In the upcoming loop, we do not yet deal with the projection_result_table indirection
   // described above.
-  auto output_segments_by_chunk = std::vector<Segments>(input_table.chunk_count());
+  auto output_segments_by_chunk = std::vector<Segments>(chunk_count);
 
-  auto forwarding_cost = std::chrono::nanoseconds{};
-  auto expression_evaluator_cost = std::chrono::nanoseconds{};
+  // Create a vector that will hold all jobs that are getting executed in parallel.
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(chunk_count);
 
-  const auto chunk_count = input_table.chunk_count();
+  const auto expression_count = expressions.size();
+
+  // Create a two-dimensional vector that saves the information if a column is nullable for every segment.
+  // This allows parallel write operation per thread. Later this data structure is aggregated
+  // to a vector with information if a column is nullable for every column.
+  auto column_is_nullable_by_chunk = std::vector<std::vector<bool>>(
+    chunk_count,
+    std::vector<bool>(expression_count, false));
+
+  // All projections that operate on chunks of sufficient size and newly generated columns will be executed after the
+  // loop by the job scheduler to parallelize the evaluation operation.
   for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
     const auto input_chunk = input_table.get_chunk(chunk_id);
     Assert(input_chunk, "Physically deleted chunk should not reach this point, see get_chunk / #1686.");
 
-    auto output_segments = Segments{expressions.size()};
+    auto output_segments = Segments{expression_count};
+    auto forwared_count = size_t{0};
```

Comment:
Typo: forwarded_count. Also, as this information is only used to check if all segments were forwarded, maybe make it `all_segments_forwarded = true` right away and simply set it to `false` in the loop?